core/consensus_test.go

```diff
@@ -121,6 +121,25 @@ func quorum(numNodes uint64) uint64 {
 	}
 }
 
+func commonHasQuorumFn(numNodes uint64) func(view *proto.View, messages []*proto.Message) bool {
+	quorum := quorum(numNodes)
+
+	return func(view *proto.View, messages []*proto.Message) bool {
+		if len(messages) > 0 {
+			switch messages[0].GetType() {
+			case proto.MessageType_PREPREPARE:
+				return len(messages) > 1
```

Comment:
The case is needed because we're subscribing to `PREPREPARE` messages here https://github.com/0xPolygon/go-ibft/blob/feature/EVM-142-move-quorum-to-backend/core/ibft.go#L204, maybe we can go with option 1

Reply:
Ok agree. Anyhow I was thinking to invoke `HasQuorum` from the `handlePrePrepare`. Just change condition accordingly, because this can be a little bit misleading :slightly_smiling_face:

Reply:
Done in https://github.com/0xPolygon/go-ibft/pull/39/commits/83353bef1c07c08cb5fea7f4302d64bcd449e870